ignite/metrics/accumulation.py

```diff
@@ -66,14 +66,19 @@ def _check_output_type(self, output: Union[Any, torch.Tensor, numbers.Number]) -
     def update(self, output: Union[Any, torch.Tensor, numbers.Number]) -> None:
         self._check_output_type(output)
 
+        if self.num_examples is None:
```

Comment:
I understand that here we bump in limits of the mypy to understand that reset is called in the constructor by the base class. However, I'd expect the checks for both vars: `num_examples` and `accumulator`. Looks a bit strange.

Reply:
the reason, why `self.num_examples` has to be checked here and not `self.accumulator` is because of those lines
https://github.com/pytorch/ignite/blob/c8ad7dc0dbfc6b62251336b57fc504b648410ea7/ignite/metrics/accumulation.py#L81-L84
`mypy` complains, that `None` and `int` have an supported opperand for `+`.

Reply:
Interesting, after I assigned scalar 0 to `num_examples`, I got a lot of `mypy` findings for `self.accumulator`, which also makes sense, but I don't understand, why they didn't pop up earlier...

Reply:
Yeah, the wisdom of `mypy` is unattainable for us humans :)